tests/python/pants_test/rules/test_filedeps.py

```diff
@@ -0,0 +1,282 @@
+    hydrated_target = self.mock_hydrated_target(target_address, {}, ())
+    
+    transitive_targets = TransitiveHydratedTargets(
+      (hydrated_target,),
+      {hydrated_target}
+    )
+    
+    self.filedeps_rule_test(
+      [target_address],
+      transitive_targets,
+      "some/target/BUILD\n"
+    )
+
+  def test_output_one_target_one_source(self):
+    target_address = self.make_build_target_address("some/target")
+    hydrated_target = self.mock_hydrated_target(target_address, {"some/file.py": "", }, ())
+    
+    transitive_targets = TransitiveHydratedTargets(
+      (hydrated_target,),
+      {hydrated_target}
+    )
+    
+    self.filedeps_rule_test(
+      [target_address],
+      transitive_targets,
+      '''some/target/BUILD
+some/file.py
+''')
+
+  def test_output_one_target_no_source_one_dep(self):
+    dep_address = self.make_build_target_address("other/target")
+    dep_target = self.mock_hydrated_target(dep_address, {"other/file.py": "", }, ())
+    
+    target_address = self.make_build_target_address("some/target")   
+    hydrated_target = self.mock_hydrated_target(target_address, {}, (dep_target,))
+    
+    transitive_targets = TransitiveHydratedTargets(
+      (hydrated_target,), 
+      OrderedSet([hydrated_target, dep_target])
+    )
+    
+    self.filedeps_rule_test(
+      [target_address],
+      transitive_targets,
+      '''some/target/BUILD
+other/target/BUILD
+other/file.py
+''')
+
+  def test_output_one_target_one_source_with_dep(self):
+    dep_address = self.make_build_target_address("other/target")
+    dep_target = self.mock_hydrated_target(dep_address, {"other/file.py": "", }, ())
+    
+    target_address = self.make_build_target_address("some/target")
+    hydrated_target = self.mock_hydrated_target(
+      target_address, 
+      {"some/file.py": "", },
+      (dep_target,)
+    )
+    
+    transitive_targets = TransitiveHydratedTargets(
+      (hydrated_target,),
+      OrderedSet([hydrated_target, dep_target])
+    )
+    
+    self.filedeps_rule_test(
+      [target_address],
+      transitive_targets,
+      '''some/target/BUILD
+some/file.py
+other/target/BUILD
+other/file.py
+''')
+
+  def test_output_multiple_targets_one_source(self):
+    target_address1 = self.make_build_target_address("some/target")
+    hydrated_target1 = self.mock_hydrated_target(target_address1, {"some/file.py": "", }, ())
+    
+    target_address2 = self.make_build_target_address("other/target")
+    hydrated_target2 = self.mock_hydrated_target(target_address2, {"other/file.py": "", }, ())
+     
+    transitive_targets = TransitiveHydratedTargets(
+      (hydrated_target1, hydrated_target2), 
+      OrderedSet([hydrated_target1, hydrated_target2])
+    )
+    
+    self.filedeps_rule_test(
+      [target_address1, target_address2],
+      transitive_targets,
+      '''some/target/BUILD
+some/file.py
+other/target/BUILD
+other/file.py
+''')
+
+  def test_outputs_multiple_targets_one_source_with_dep(self):
```

Comment:
Possibly update the graph in the comment to reflect that `target2` also has a dep.